spec/requests/students_request_spec.rb

```diff
@@ -0,0 +1,255 @@
+  let(:now_student) { nil }
+  let(:new_user) {create(:user)}
+
+  describe 'GET #index' do
+    before(:each) do
+      sign_in current_user
+      get admin_students_path
+    end
+
+    context 'with logged-in admin' do
+      let(:current_user) {admin}
+      it 'index status' do
+        expect(response).to have_http_status(:success)
+      end
+      it 'render index view' do
+        is_expected.to render_template :index
+      end
+    end
+
+    context 'with logged-in user' do
+      let(:current_user) { user }
+      it 'index status' do
+        expect(response).to redirect_to root_path
+      end
+    end
+
+    context "not logged-in user" do
+      let(:current_user) { :not_user }
+      it { expect(response.status).to eq(302) }
+      it { expect(response).to redirect_to new_user_session_path }
+    end
+  end
+
+  describe 'GET #show' do
+    before(:each) do
+      sign_in current_user
+      get admin_student_path(id: student.id)
+    end
+
+    context 'with logged-in admin' do
+      let(:current_user) { admin }
+      it 'show status' do
+        expect(response).to have_http_status(:success)
+      end
+      it 'should show student' do
+        is_expected.to render_template :show
+      end
+    end
+
+    context 'does not exist student' do
+      let(:current_user) { admin }
+      it { expect { get admin_student_path(id: invalid_id) }.to raise_error(ActiveRecord::RecordNotFound) }
+    end
+
+    context 'with logged-in user' do
+      let(:current_user) { user }
+      it 'redirects to root-path' do
+        expect(response).to redirect_to root_path
+      end
+    end
+
+    context "not logged-in user" do
+      let(:current_user) {:not_user}
+      it { expect(response.status).to eq(302) }
+      it { expect(response).to redirect_to new_user_session_path }
+    end
+  end
+
+  describe 'GET #new' do
+    before(:each) do
+      sign_in current_user
+      get new_admin_student_path
+    end
+
+    context 'with logged-in admin' do
+      let(:current_user) {admin}
+      it 'create a new Student' do
+        expect(assigns(:student)).to be_a_new(Student)
+      end
+      it 'renders the :new template' do
+        expect(response).to render_template :new
+      end
+    end
+
+    context 'with logged-in user' do
+      let(:current_user) {user}
+      it 'redirects to root-path' do
+        expect(response).to redirect_to root_path
+      end
+    end
+
+    context "not logged-in user" do
+      let(:current_user) { :not_user }
+      it { expect(response.status).to eq(302) }
+      it { expect(response).to redirect_to new_user_session_path }
+    end
+  end
+
+  describe 'CREATE #create' do
+    context 'with logged-in admin' do
+      context 'with valid data' do
+        let(:user) {create(:user)}
+        let(:grade) {create(:grade)}
+        let(:parent) {create(:parent)}
+        let(:student) {create(:student)}
+        before do
+          sign_in admin
+        end
+
+        it { expect do
```

Comment:
what are you testing here?

Reply:
I don't think student_params should have student_id

Reply:
I was testing creating of student. here i have made mistake. I will fix it.